File: WorkoutTracker.Domain/Entities/Workouts/WorkoutRecord.cs
using WorkoutTracker.Domain.Base;
using WorkoutTracker.Domain.Entities.Exercises.ExerciseGroups;

namespace WorkoutTracker.Domain.Entities.Workouts;

public class WorkoutRecord : IDbModel
{
    public long Id { get; set; }
    public TimeSpan Time { get; set; }
    public DateTime Date { get; set; }

    public string UserId { get; set; } = null!;

    public long WorkoutId { get; set; }
    public Workout? Workout { get; set; }

    public IEnumerable<ExerciseRecordGroup> ExerciseRecordGroups { get; set; } = null!;
}

File: WorkoutTracker/src/app/workouts/models/workout-record.ts
import { Exercise } from "src/app/exercises/models/exercise";
import { DbModel } from "src/app/shared/models/base/db-model";
import { ExerciseRecordGroup } from "src/app/shared/models/exercises/exercise-record-group";
import { ModelWeight } from "src/app/shared/models/model-weight";
import { TimeSpan } from "src/app/shared/models/time-span";

export interface WorkoutRecord extends DbModel {
    time: TimeSpan;
    date: Date;
    weight: ModelWeight;

    workoutId: number;
    workoutName: string;

    exercises: Exercise[];
    exerciseRecordGroups: ExerciseRecordGroup[];
}